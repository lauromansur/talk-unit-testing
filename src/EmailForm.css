.email__form {
  background-color: rgba(255, 255, 255, .1);
  padding: 24px;
}

.form__control {
  margin-bottom: 12px;
}

label {	
  color: #61dafb;
  display: block;
  font-size: calc(10px + 1vmin);
  margin-bottom: 6px;
  text-align: left;
}

button {
  background-color: #61dafb;
  color: #ffffff;
  cursor: pointer;
  font-weight: bold;
  padding: 12px 24px;
  text-transform: uppercase;
}

textarea {
  height: 50px;
  max-height: 300px;
  resize: vertical;
}

input, textarea {
  padding: 12px;
  width: 220px;
}

input {
  max-height: 35px;
}

button, input, textarea {
  border-color: transparent;
  border-radius: 6px;
  border-width: 0;
}

.form__actions {
  justify-content: center;
  display: flex;
}

.button__icon {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

svg {
  height: 18px;
  margin-left: 6px;
  width: 18px;
}
