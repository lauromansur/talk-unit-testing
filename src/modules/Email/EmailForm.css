.email__form {
  background-color: rgba(255, 255, 255, .1);
  padding: 24px;
}

button {
  background-color: #61dafb;
  color: #ffffff;
  cursor: pointer;
  font-weight: bold;
  padding: 12px 24px;
  text-transform: uppercase;
}

.form__actions {
  justify-content: center;
  display: flex;
}

.button__icon {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

svg {
  height: 18px;
  margin-left: 6px;
  width: 18px;
}
