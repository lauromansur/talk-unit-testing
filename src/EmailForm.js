import React, { Component } from 'react';

import './EmailForm.css';

class EmailForm extends Component {
  render() {
    return (
      <div className="email-form">
        <p>
          Email form component.
        </p>
      </div>
    );
  }
}

export default EmailForm;
