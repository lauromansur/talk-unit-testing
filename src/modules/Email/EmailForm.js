import React, { Component } from 'react';

import EmailService from './EmailService';
import FormControl from '../../components/FormControl/FormControl';
import { ReactComponent as PaperPlaneSolidIcon } from '../../icons/paper-plane-solid.svg';

import './EmailForm.css';

class EmailForm extends Component {
  state = {
    from: '',
    to: '',
    subject: '',
    message: ''
  }

  setValue(event) {
    const { name, value } = event.target;

    this.setState({ [name]: value });
  }

  sendEmail(event) {
    event.preventDefault();

    try {
      const response = EmailService.sendEmail(this.state);

      if (response) {
        this.clearForm();
      }
    } catch(error) {
      console.error(error);
    }
  }

  clearForm() {
    this.setState({
      from: '',
      to: '',
      subject: '',
      message: ''
    });
  }

  render() {
    const { from, to, subject, message } = this.state;

    return (
      <form className="email__form">
        <FormControl
          id="from"
          name="from"
          labelText="From:"
          onChange={this.setValue.bind(this)}
          value={from}
        />

        <FormControl
          id="to"
          name="to"
          labelText="To:"
          onChange={this.setValue.bind(this)}
          value={to}
        />

        <FormControl
          id="subject"
          name="subject"
          labelText="Subject:"
          onChange={this.setValue.bind(this)}
          value={subject}
        />

        <FormControl
          textarea
          id="message"
          name="message"
          labelText="Message:"
          onChange={this.setValue.bind(this)}
          value={message}
        />

        <div className="form__actions">
          <button
            type="submit"
            className="button__icon"
            onClick={this.sendEmail.bind(this)}
          >
            Send
            <PaperPlaneSolidIcon />
          </button>
        </div>
      </form>
    );
  }
}

export default EmailForm;
