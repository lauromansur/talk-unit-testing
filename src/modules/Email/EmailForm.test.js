import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import EmailForm from './EmailForm';

describe('EmailForm', () => {
  test('renders component tree correctly (without enzyme)', () => {
    const tree = renderer
    .create(<EmailForm />)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders component tree correctly (with enzyme)', () => {
    const tree = shallow(<EmailForm />);

    expect(tree).toMatchSnapshot();
  });
});
