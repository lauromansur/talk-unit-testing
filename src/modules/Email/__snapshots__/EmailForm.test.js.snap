// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailForm renders component tree correctly (with enzyme) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormControl
          autoComplete="off"
          id="from"
          labelText="From:"
          name="from"
          onChange={[Function]}
          value=""
        />,
        <FormControl
          autoComplete="off"
          id="to"
          labelText="To:"
          name="to"
          onChange={[Function]}
          value=""
        />,
        <FormControl
          autoComplete="off"
          id="subject"
          labelText="Subject:"
          name="subject"
          onChange={[Function]}
          value=""
        />,
        <FormControl
          autoComplete="off"
          id="message"
          labelText="Message:"
          name="message"
          onChange={[Function]}
          textarea={true}
          value=""
        />,
        <div
          className="form__actions"
        >
          <button
            className="button__icon"
            onClick={[Function]}
            type="submit"
          >
            Send
            <ReactComponent />
          </button>
        </div>,
      ],
      "className": "email__form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoComplete": "off",
          "id": "from",
          "labelText": "From:",
          "name": "from",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoComplete": "off",
          "id": "to",
          "labelText": "To:",
          "name": "to",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoComplete": "off",
          "id": "subject",
          "labelText": "Subject:",
          "name": "subject",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoComplete": "off",
          "id": "message",
          "labelText": "Message:",
          "name": "message",
          "onChange": [Function],
          "textarea": true,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="button__icon"
            onClick={[Function]}
            type="submit"
          >
            Send
            <ReactComponent />
          </button>,
          "className": "form__actions",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Send",
              <ReactComponent />,
            ],
            "className": "button__icon",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": Array [
            "Send",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormControl
            autoComplete="off"
            id="from"
            labelText="From:"
            name="from"
            onChange={[Function]}
            value=""
          />,
          <FormControl
            autoComplete="off"
            id="to"
            labelText="To:"
            name="to"
            onChange={[Function]}
            value=""
          />,
          <FormControl
            autoComplete="off"
            id="subject"
            labelText="Subject:"
            name="subject"
            onChange={[Function]}
            value=""
          />,
          <FormControl
            autoComplete="off"
            id="message"
            labelText="Message:"
            name="message"
            onChange={[Function]}
            textarea={true}
            value=""
          />,
          <div
            className="form__actions"
          >
            <button
              className="button__icon"
              onClick={[Function]}
              type="submit"
            >
              Send
              <ReactComponent />
            </button>
          </div>,
        ],
        "className": "email__form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": "off",
            "id": "from",
            "labelText": "From:",
            "name": "from",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": "off",
            "id": "to",
            "labelText": "To:",
            "name": "to",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": "off",
            "id": "subject",
            "labelText": "Subject:",
            "name": "subject",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": "off",
            "id": "message",
            "labelText": "Message:",
            "name": "message",
            "onChange": [Function],
            "textarea": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button__icon"
              onClick={[Function]}
              type="submit"
            >
              Send
              <ReactComponent />
            </button>,
            "className": "form__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Send",
                <ReactComponent />,
              ],
              "className": "button__icon",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": Array [
              "Send",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`EmailForm renders component tree correctly (without enzyme) 1`] = `
<form
  className="email__form"
>
  <div
    className="form__control"
  >
    <label
      htmlFor="from"
    >
      From:
    </label>
    <input
      autoComplete="off"
      id="from"
      name="from"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="form__control"
  >
    <label
      htmlFor="to"
    >
      To:
    </label>
    <input
      autoComplete="off"
      id="to"
      name="to"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="form__control"
  >
    <label
      htmlFor="subject"
    >
      Subject:
    </label>
    <input
      autoComplete="off"
      id="subject"
      name="subject"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="form__control"
  >
    <label
      htmlFor="message"
    >
      Message:
    </label>
    <textarea
      autoComplete="off"
      id="message"
      name="message"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="form__actions"
  >
    <button
      className="button__icon"
      onClick={[Function]}
      type="submit"
    >
      Send
      <svg>
        paper-plane-solid.svg
      </svg>
    </button>
  </div>
</form>
`;
